lib/internal/util.js

```diff
@@ -95,7 +95,7 @@ function isError(e) {
   // An error could be an instance of Error while not being a native error
   // or could be from a different realm and not be instance of Error but still
   // be a native error.
-  return isNativeError(e) || e instanceof Error;
+  return Error.isError ? Error.isError(e) : isNativeError(e) || e instanceof Error;
```

Comment:
Do we need to ensure backward compatibility?

Reply:
`Error.isError` should be the entirety of what `isNativeError` does, and isNativeError should no longer be needed nor `instanceof` used.

Wouldn't it make more sense to just remove `isError` entirely, and use the `ErrorIsError` primordial directly, and add "don't land" labels on this PR?

Reply:
+1, @ljharb

Reply:
Got it! Do you think it is okay to continue with this PR and issue or would it be better to create a new one?

Reply:
seems fine to me to adapt this PR to do that.

Reply:
We still want to match errors that are not regular ones. The wild has many examples of these, so we should keep instanceof.